Cannot stop App Engine local server in certain case

I think this is likely a dev_appserver1 issue.

Steps to reproduce:
- Create a new HelloWorld project.
- Change the text of `<servlet-name>` in `web.xml` to something else (e.g., to `NonExistingHelloAppEngine`).
   - This is a semantic configuration error, so deploying or running this app itself shouldn't be a problem.
   - BTW, our validation will correctly mark an error that the `HelloAppEngine` servlet does not exist, which is nice.
- _Run As_ > _App Engine_. The browser will show 503 SERVICE_UNAVAILABLE, and the console will also show the following stack trace:
   ```
   WARNING: Failed startup of context com.google.appengine.tools.development.DevAppEngineWebAppContext@1023b85{/,/ ... /workspace_qual/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/A}
   java.lang.IllegalStateException: No such servlet: HelloAppEngine
   	at org.mortbay.jetty.servlet.ServletHandler.updateMappings(ServletHandler.java:1026)
   ```
- Try stopping the server. The server won't stop. You can see the following Eclipse log:
```
WARNING: Error terminating server: Error connecting to http://localhost:8080/_ah/admin/quit
com.google.cloud.tools.appengine.api.AppEngineException: Error connecting to http://localhost:8080/_ah/admin/quit
        at com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDevServer1.stop(CloudSdkAppEngineDevServer1.java:166)
        at com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEngineServerBehaviour.stop(LocalAppEngineServerBehaviour.java:139)
        at org.eclipse.wst.server.core.internal.Server.stopImpl2(Server.java:3688)
        at org.eclipse.wst.server.core.internal.Server.stopImpl(Server.java:3645)
        at org.eclipse.wst.server.core.internal.Server$StopJob.run(Server.java:403)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: java.io.IOException: Server returned HTTP response code: 503 for URL: http://localhost:8080/_ah/admin/quit
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1876)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
        at com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDevServer1.stop(CloudSdkAppEngineDevServer1.java:158)
        ... 5 more
```

So, the dev_appserver1 responds with 503 when we connect to `http://localhost:8080/_ah/admin/quit`.

--- a/plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java
@@ -154,17 +154,22 @@ public class LocalAppEngineServerBehaviour extends ServerBehaviourDelegate
         devServer.stop(stopConfig);
       } catch (AppEngineException ex) {
         logger.log(Level.WARNING, "Error terminating server: " + ex.getMessage(), ex); //$NON-NLS-1$
+        terminate();
       }
     } else {
       // we've already given it a chance
-      logger.info("forced stop: destroying associated processes"); //$NON-NLS-1$
-      if (devProcess != null) {
-        devProcess.destroy();
-        devProcess = null;
-      }
-      devServer = null;
-      setServerState(IServer.STATE_STOPPED);
+      terminate();
     }
+  }
+
+  private void terminate() {
+    logger.info("forced stop: destroying associated processes"); //$NON-NLS-1$
+    if (devProcess != null) {
+      devProcess.destroy();
+      devProcess = null;
+    }
+    devServer = null;
+    setServerState(IServer.STATE_STOPPED);
   }
 
 

